Dockerfile

```diff
@@ -5,10 +5,28 @@
 ARG AVALANCHEGO_NODE_IMAGE
 
 # ============= Compilation Stage ================
-FROM golang:1.22.8-bullseye AS builder
+FROM --platform=$BUILDPLATFORM golang:1.22.8-bullseye AS builder
 
 WORKDIR /build
 
+
+ARG TARGETPLATFORM
```

Comment:
When do we need the few instructions below? Ideally have any reference to TARGETPLATFORM the latest possible since this will trigger image builder to parallelize for each target platform. It's a bit implicit and not well documented, but I remember seeing docker building being much slower because of this when cross building (i.e. go mod download duplicated for N platforms)

I suppose this should be right before the last `RUN` instruction.

Reply:
I think this is a built-in arg and referencing should not impact anything.